src/plugins/persistence/couch/CouchSearchProvider.js

```diff
@@ -67,18 +71,45 @@ class CouchSearchProvider {
     }
   }
 
-  searchForObjects(query, abortSignal) {
-    const filter = {
-      selector: {
-        model: {
-          name: {
-            $regex: `(?i)${query}`
+  #isOptimizedSearchByNameSupported() {
+    if (this.#isSearchByNameViewDefined === undefined) {
```

Comment:
i guess you might as well do a single `return` here as well

Reply:
How dare you turn my own rules against me.

Reply:
Fixed!